### PR TITLE
15.0 mig openupgrade framework

### PR DESCRIPTION
**Pull request by legalsylvain**
Hi all. 

I just executed meld on odoo subfolder to check the differences, to guess if the openupgrade_framework was requiring changes. 
At first sight : very fiew changes. 

**some changes related to base framework** (for information)

-  [IMP] core: introduce api.ondelete decorator  (https://github.com/odoo/odoo/commit/1c8a95809852baefa851985507a07ac7f6465dc7)
-  [IMP] core: allow set [value] in selection ondelete. (https://github.com/odoo/odoo/commit/cc768f39ebd71b89da8a0a2dc680f6725e2a4359)
-  [IMP] base: remove get_object helper (https://github.com/odoo/odoo/commit/fd67339a3ef754553bc787965fe8cedd720ce618)

**Some changes related to OpenUpgrade framework**

- lot of changes in ``_check_xml``. Specially https://github.com/odoo/odoo/commit/46f1912c9b9ee9767d6f32b725c691509dc5e5b2. Nothing to do in my opinion 

- lot of changes in ``_post_process_view``. Here https://github.com/odoo/odoo/commit/331b54770e21a8dd4617d05f0bee2cc0383ba63b. Nothing to do in my opinion 

- ``handle_view_error`` has been splitted in two functions here https://github.com/odoo/odoo/commit/9ae8c1a7c984183d6d1bd131ffcd9d4269c19a1f. **TODO**

### CODE REVIEW

File: openupgrade_scripts/scripts/base/15.0.1.3/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    Don't request an env for the base pre migration as flushing the env in
+    odoo/modules/registry.py will break on the 'base' module not yet having
+    been instantiated.
+    """
+    if "openupgrade_framework" not in tools.config["server_wide_modules"]:
+        logging.error(
+            "openupgrade_framework is not preloaded. You are highly "
+            "recommended to run the Odoo with --load=openupgrade_framework "
+            "when migrating your database."
+        )
```
Comment (jcdrubay) [inline openupgrade_scripts/scripts/base/15.0.1.3/pre-migration.py:68]: It seems like you will need something like this:

```suggestion
        )
        openupgrade.rename_xmlids(cr, [("mail.icp_mail_catchall_alias", "base.icp_mail_catchall_alias")])
```

**Comment by oerp-odoo** [comment]
Whats the status of this PR? Is it possible to get it merged soon?

**Comment by sutidor** [comment]
Nice. Looking forward to see this done.

**Comment by legalsylvain** [comment]
Hi. This PR is WIP.
No time immediately. Help welcome.

regards.

File: openupgrade_scripts/scripts/base/15.0.1.3/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        openupgrade.rename_xmlids(cr, [("mail.icp_mail_catchall_alias", "base.icp_mail_catchall_alias")])
```
Comment (royle-vietnam) [inline openupgrade_scripts/scripts/base/15.0.1.3/pre-migration.py]: Line too long and correct indentation
```suggestion
    openupgrade.rename_xmlids(
        cr,
        [
            ("mail.icp_mail_catchall_alias", "base.icp_mail_catchall_alias"),
            ("mail.icp_mail_bounce_alias", "base.icp_mail_bounce_alias"),
        ],
    )
```

File: openupgrade_framework/odoo_patch/odoo/addons/base/models/ir_ui_view.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                self.model,
+            )
+            self.write({"active": False})
+
+
+def _postprocess_view(self, node, model, validate=True, editable=True):
+    """Don't validate views, _raise_view_error is mutted"""
+    return View._postprocess_view._original_method(
+        self, node, model, validate=False, editable=editable
+    )
```
Comment (royle-vietnam) [inline openupgrade_framework/odoo_patch/odoo/addons/base/models/ir_ui_view.py]: @legalsylvain correct number of args
```suggestion
def _postprocess_view(self, node, model, editable=True):
    """Don't validate views, _raise_view_error is mutted"""
    return View._postprocess_view._original_method(
        self, node, model, editable=editable
    )
```

**Comment by royle-vietnam** [comment]
@legalsylvain Looking forward to see this done. Could you make a todo list to follow / help?

**Comment by legalsylvain** [comment]
@royleviindoo. 
Thanks for the ping.

I haven't worked on this RP for a while, so I'm not sure! 
I'll plan to work on it again, the week of May 9, and let you know what it is.

Have a good weekend.

**Comment by jcdrubay** [comment]
@royleviindoo FYI, I am currently working on this fork https://github.com/jcdrubay/OpenUpgrade/tree/15.0-add-xml-ids-moved-from-mail-to-base

**Comment by royle-vietnam** [comment]
> @royleviindoo FYI, I am currently working on this fork https://github.com/jcdrubay/OpenUpgrade/tree/15.0-add-xml-ids-moved-from-mail-to-base

Hi @jcdrubay , I looked at your repository, but didn't see any difference
FYI, Our company is currently working on repo https://github.com/Viindoo/OpenUpgrade/blob/15.0/docsource/modules140-150.rst

**Comment by legalsylvain** [comment]
@royleviindoo : you can review the diff hère :  

https://github.com/legalsylvain/OpenUpgrade/pull/4

**Comment by legalsylvain** [comment]
Hi @jcdrubay. CI is now green. Thanks ! 

just tested a migration from 14 to 15 with that up-to-date branch (``account`` + ``web_responsive`` installed).
- migration passed
- V15 is available

Some remaining points : 
1. I have a warning
```
2022-05-13 13:33:52,958 1 WARNING test__openupgrade__14__15_a py.warnings: /usr/lib/python3.9/contextlib.py:117: DeprecationWarning: Since Odoo 15.0, Environment.manage() is useless.

  File "/odoo_env/src/openupgrade/openupgrade_framework/odoo_patch/odoo/modules/migration.py", line 17, in migrate_module
    MigrationManager.migrate_module._original_method(self, pkg, stage)
  File "/odoo_env/src/odoo/odoo/modules/migration.py", line 180, in migrate_module
    migrate(self.cr, installed_version)
  File "/usr/local/lib/python3.9/dist-packages/openupgradelib/openupgrade.py", line 1992, in wrapped_function
    contextmanagers.enter_context(api.Environment.manage())
```
Something should be fixed in openupgradelib project.

2. I have a lot of logs, related to renaming : 
```2022-05-13 13:33:44,614 1 INFO test__openupgrade__14__15_a OpenUpgrade: XMLID l10n_ec.state_ec_1 not found when renaming to base.state_ec_01```

Fixed here : https://github.com/OCA/OpenUpgrade/pull/2967/commits/aebb0d2c2b243f12c57c0ae007600932786e8de7

3. Did you planned to finish ``base`` migration ? it seems there are some points present in the analysis files that are not handled ?  (just a question)

**Comment by jcdrubay** [comment]
I would be happy to continue the work on base module.
It would help me a lot if you could pinpoint some of the remaining tasks to do as I did not realize something was missing.

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Pull request merged**